hazelcast/src/test/java/com/hazelcast/internal/adapter/DataStructureMethodsTest.java

```diff
@@ -0,0 +1,57 @@
+package com.hazelcast.internal.adapter;
+
+import com.hazelcast.internal.adapter.DataStructureAdapter.DataStructureMethods;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.junit.Assert.fail;
+
+/**
+ * Checks that {@link DataStructureMethods} contains unique method signatures which are defined in {@link DataStructureAdapter}.
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class DataStructureMethodsTest {
+
+    private static final Class<?> DATA_STRUCTURE_ADAPTER = DataStructureAdapter.class;
+    private static final Class<?> DATA_STRUCTURE_METHODS = DataStructureMethods.class;
+
+    @Test
+    public void testThatAllDataStructureMethodsAreFoundAndReferencedJustOnce() {
+        String adapterClassName = DATA_STRUCTURE_ADAPTER.getSimpleName();
+        String methodsClassName = DATA_STRUCTURE_METHODS.getSimpleName();
+
+        Class<?> dataStructureAdapterImpl = new ByteBuddy()
+                .subclass(DATA_STRUCTURE_ADAPTER)
+                .make()
+                .load(getClass().getClassLoader(), ClassLoadingStrategy.Default.WRAPPER)
+                .getLoaded();
+
+        Map<Method, DataStructureMethods> knownMethods = new HashMap<Method, DataStructureMethods>();
+        for (DataStructureMethods method : DataStructureMethods.values()) {
+            try {
+                Method adapterMethod = dataStructureAdapterImpl.getMethod(method.getMethodName(), method.getParameterTypes());
```

Comment:
Stupid me... I first tried to get an implementation whose methods could be invoked, so I had a look at Byte Buddy to have a real instance. But for the signature checking this was not needed, but I left Byte Buddy in the code. I simplified the test now. Very good finding, thanks!